File: browser/extensions/webcompat/experiment-apis/sharedPreferences.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

"use strict";

/* global ExtensionAPI, Services, XPCOMUtils */

XPCOMUtils.defineLazyModuleGetters(this, {
  Services: "resource://gre/modules/Services.jsm",
  SharedPreferences: "resource://gre/modules/SharedPreferences.jsm",
});

this.sharedPreferences = class extends ExtensionAPI {
  getAPI(context) {
    return {
      sharedPreferences: {
        async setCharPref(name, value) {
          if (!Services.androidBridge || !Services.androidBridge.isFennec) {
            return;
          }
          SharedPreferences.forApp().setCharPref(name, value);
        },
        async setBoolPref(name, value) {
          if (!Services.androidBridge || !Services.androidBridge.isFennec) {
            return;
          }
          SharedPreferences.forApp().setBoolPref(name, value);
        },
      },
    };
  }
};

File: mobile/android/fenix/app/src/test/java/org/mozilla/fenix/ext/SharedPreferences.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package org.mozilla.fenix.ext

import android.content.SharedPreferences
import androidx.core.content.edit

/**
 * Clear everything in shared preferences and commit changes immediately.
 */
fun SharedPreferences.clearAndCommit() = this.edit(commit = true) { clear() }
